discovery-provider/src/queries/get_health.py

```diff
@@ -110,27 +112,14 @@ def get_health(args, use_redis_cache=True):
 
     latest_block_num = None
     latest_block_hash = None
-
-    # Get latest web block info
-    if use_redis_cache:
-        stored_latest_block_num = redis.get(latest_block_redis_key)
-        if stored_latest_block_num is not None:
-            latest_block_num = int(stored_latest_block_num)
-
-        stored_latest_blockhash = redis.get(latest_block_hash_redis_key)
-        if stored_latest_blockhash is not None:
-            latest_block_hash = stored_latest_blockhash.decode("utf-8")
-
-    if latest_block_num is None or latest_block_hash is None:
-        latest_block = web3.eth.getBlock("latest", True)
-        latest_block_num = latest_block.number
-        latest_block_hash = latest_block.hash.hex()
-
     latest_indexed_block_num = None
     latest_indexed_block_hash = None
 
-    # Get latest indexed block info
     if use_redis_cache:
+        # get latest blockchain state from redis cache, or fallback to chain if None
```

Comment:
I'm a little worried about this change causing health check to be significantly slower some of the time (since we're manually clearing the key). The client's heavily depend on health check being as fast as possible for selection.

Reply:
Maybe we should have sep. keys for health vs. indexing?

Reply:
this is actually never used in indexing. the reason we added these keys was only for health check

as for health check latency, it's possible that an uncached value could cause the time to spike, but the tradeoff here is latency vs correctness. one tradeoff i'm willing to make is increase the ttl of this to 2x+1 of the indexing window so like 11 seconds. that way we should be no more than 3 blocks behind in terms of accuracy but if indexing hasn't fired off and set the new value in redis, we'll get an accurate block diff

Reply:
also i think until we benchmark on a prod node we can't really test the latency effects

Reply:
gotcha. i kinda see more value in the monitoring method that was suggested-- have this done independently of the health check behind the scenes make the most sense to me. we could make the calculation happen every 5s still or something. im ok to ship this now if you think the benefit from this correctness outways the potential incurred cost.

Reply:
i think the benefits here are more higher level. certain discovery nodes have a tendency to oscillate from like 0 to 20 blocks behind and back to 0. effectively this represents like 90 seconds of falling behind and then catching up, which i think this may also contribute to content nodes timing out in the middlewares. however if the true block diff is exposed, when we make requests through libs (or cnode via libs) to the discovery node, if the block diff is accurate, _makeRequest in the libs discovery service be much quicker and more accurate to move to run discovery reselection and move to another node

Reply:
yeah, while i do think it would be better totally async from the health check, this makes a lot of sense. if your hunch that this will help the middleware situation is correct, we should def ship this asap